src/chat.rs

```diff
@@ -466,6 +467,71 @@ impl ChatId {
         }
     }
 
+    /// Get autodelete timer value in seconds.
+    pub async fn get_autodelete_timer(self, context: &Context) -> u32 {
+        context
+            .sql
+            .query_get_value(
+                context,
+                "SELECT autodelete_timer FROM chats WHERE id=?;",
+                paramsv![self],
+            )
+            .await
+            .unwrap_or_default()
+    }
+
+    /// Set autodelete timer value without sending a message.
+    ///
+    /// Used when a message arrives indicating that someone else has
+    /// changed the timer value for a chat.
+    pub(crate) async fn inner_set_autodelete_timer(
+        self,
+        context: &Context,
+        timer: u32,
+    ) -> Result<(), Error> {
+        ensure!(!self.is_special(), "Invalid chat ID");
+
+        context
+            .sql
+            .execute(
+                "UPDATE chats
+             SET autodelete_timer=?
+             WHERE id=?;",
+                paramsv![timer, self],
+            )
+            .await?;
+        Ok(())
+    }
+
+    /// Set autodelete timer value in seconds.
+    ///
+    /// If timer value is 0, disable autodelete timer.
+    pub async fn set_autodelete_timer(self, context: &Context, timer: u32) -> Result<(), Error> {
+        if timer == self.get_autodelete_timer(context).await {
+            return Ok(());
+        }
+        self.inner_set_autodelete_timer(context, timer).await?;
+        let mut msg = Message::new(Viewtype::Text);
+        msg.text = Some(
+            context
+                .stock_system_msg(
+                    StockMessage::MsgAutodeleteTimerChanged,
+                    timer.to_string(),
+                    "",
+                    0,
+                )
+                .await,
+        );
+        msg.param.set_cmd(SystemMessage::AutodeleteTimerChanged);
+        if let Err(err) = send_msg(context, self, &mut msg).await {
+            warn!(
+                context,
+                "Failed to send a message about autodelete timer change: {:?}", err
+            );
```

Comment:
I will change this to `error!`, because it should not normally happen, so we will get bug reports if something goes wrong.

But if the system message is not sent, it is not a serious problem: the timer is sent in every message anyway. So you will not send a message without a timer accidentally.

Reply:
Oh, I see.  Perhaps it's worth adding a comment saying that this will be sent again with each next message anyway and therefore simply reporting the error is enough.